src/cplscheme/CompositionalCouplingScheme.cpp

```diff
@@ -73,13 +73,15 @@ bool CompositionalCouplingScheme::isInitialized() const
   return isInitialized;
 }
 
-void CompositionalCouplingScheme::addComputedTime(double timeToAdd)
+bool CompositionalCouplingScheme::addComputedTime(double timeToAdd)
 {
   PRECICE_TRACE(timeToAdd);
 
+  bool isAtWindowEnd = false;
   for (const auto scheme : schemesToRun()) {
-    scheme->addComputedTime(timeToAdd);
+    isAtWindowEnd |= scheme->addComputedTime(timeToAdd); // @todo should be &= ?
```

Comment:
@fsimonis @uekerman : Any opinion on this? I'm a bit puzzled, because I would expect `&=`, but with `&=` I got failures of intergration tests in #1520. In this PR, it does not matter (so we are probably missing a test).

Reply:
I guess that we anyway don't support composition of coupling schemes with different time windows. Correct @fsimonis ?
Could then also become an assertion here. All true or all false.